rapid_pro_tools/rapid_pro_client.py

```diff
@@ -37,28 +84,174 @@ def get_raw_runs_for_flow_id(self, flow_id, range_start_inclusive=None, range_en
             flow=flow_id, after=range_start_inclusive, before=range_end_inclusive).all(retry_on_rate_exceed=True)
         print(f"Fetched {len(raw_runs)} runs")
 
+        if raw_export_log is not None:
+            print(f"Logging {len(raw_runs)} fetched runs...")
+            json.dump([contact.serialize() for contact in raw_runs], raw_export_log)
+            raw_export_log.write("\n")
+            print(f"Logged fetched contacts")
+        else:
+            print("Not logging the raw export (argument 'raw_export_log' was None)")
+
         # Sort in ascending order of modification date
         raw_runs = list(raw_runs)
         raw_runs.reverse()
 
         return raw_runs
 
-    def get_raw_contacts(self):
-        print("Fetching all raw contacts...")
-        raw_contacts = self.rapid_pro.get_contacts().all(retry_on_rate_exceed=True)
+    def get_raw_contacts(self, range_start_inclusive=None, range_end_exclusive=None, raw_export_log=None):
+        """
+        Gets the raw contacts from RapidPro.
+        
+        :param range_start_inclusive: Start of the date-range to download contacts from. If set, only downloads contacts
```

Comment:
Right - so this is modified. 

I think we might need to put this in the variable names, there are few too many different interpretations.

Reply:
Considered contacts_modified_range_inclusive

It's getting a bit long winded, but important I think

Reply:
renamed to last_modified_after_inclusive etc.